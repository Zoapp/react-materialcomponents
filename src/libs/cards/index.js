/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import CardActions from "./actions";
import Card from "./card";
import CardMedia from "./media";
import CardText from "./text";
import CardPrimary from "./primary";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";

export {
  Card,
  CardActions,
  CardMedia,
  CardText,
  CardPrimary,
  CardTitle,
  CardSubtitle,
};
