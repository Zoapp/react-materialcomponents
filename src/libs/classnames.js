/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
  Rewritten to ES7 by Mik BRY
*/
const ClassNames = (args) => {
  const classes = [];
  args.forEach((arg) => {
    if (arg) {
      const argType = typeof arg;
      if (argType === "string" || argType === "number") {
        classes.push(arg);
      } else if (Array.isArray(arg) && arg.length) {
        const inner = ClassNames(arg);
        if (inner) {
          classes.push(inner);
        }
      } else if (argType === "object") {
        Object.keys(arg).forEach((key) => {
          if (arg[key]) {
            classes.push(key);
          }
        });
      }
    }
  });
  return classes.join(" ");
};

export default ClassNames;
