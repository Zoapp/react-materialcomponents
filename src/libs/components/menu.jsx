/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import Rmdc from "../";
/*
mdc-simple-menu
See:
https://material.io/components/web/catalog/menus/
https://material-components-web.appspot.com/simple-menu.html

TODO : all

*/
export default class Menu extends Component {
  render() {
    const {
      className, disabled, ...props
    } = this.props;
    let classes = "mdc-simple-menu";
    if (className) {
      classes += ` ${className}`;
    }
    const d = {};
    if (disabled) {
      d.disabled = "disabled";
    }
    const element = (
      <div className={classes} tabIndex="-1" {...d} >
        <ul className="mdc-simple-menu__items mdc-list" role="menu" aria-hidden="true">
          <li className="mdc-list-item" role="menuitem" tabIndex="0">
            A Menu Item
          </li>
          <li className="mdc-list-item" role="menuitem" tabIndex="0">
            Another Menu Item
          </li>
        </ul>
      </div>);
    return Rmdc.render(element, props);
  }
}

Menu.defaultProps = {
  className: null,
  disabled: false,
};

Menu.propTypes = {
// React component props
  className: PropTypes.string,
  disabled: PropTypes.bool,
};
