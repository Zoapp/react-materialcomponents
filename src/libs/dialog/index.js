/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Dialog from "./dialog";
import DialogBody from "./body";
import DialogFooter from "./footer";
import DialogHeader from "./header";

export { Dialog, DialogBody, DialogFooter, DialogHeader };
