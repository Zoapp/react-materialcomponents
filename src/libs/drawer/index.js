import Drawer from "./drawer";
import DrawerContent from "./content";
import DrawerHeader from "./header";

export { Drawer, DrawerContent, DrawerHeader };
