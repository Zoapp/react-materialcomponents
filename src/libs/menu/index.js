/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import MenuAnchor from "./anchor";
import MenuItem from "./item";
import MenuDivider from "./divider";

import Menu from "./menu";

export { Menu, MenuAnchor, MenuItem, MenuDivider };
