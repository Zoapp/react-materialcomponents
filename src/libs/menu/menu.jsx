/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component, Children } from "react";
import PropTypes from "prop-types";
import Zrmc from "../";

/**
 * mdc-simple-menu
 * See:
 * https://material.io/components/web/catalog/menus/
 * https://material-components-web.appspot.com/simple-menu.html
 *
 * TODO:
 * - animation
 * - autoposition
 * - autofocus
 * see https://github.com/material-components/material-components-web/blob/master/packages/mdc-menu/foundation.js
 *
 */

const MDC_SIMPLE_MENU = "mdc-menu";

export default class Menu extends Component {
  constructor(props) {
    super(props);
    this.selectedRef = null;
  }

  componentDidMount() {
    this.updateFocus();
  }

  componentDidUpdate() {
    this.updateFocus();
  }

  handleOnSelected = (child, index) => {
    if (this.props.onSelected) {
      this.props.onSelected(child, index);
    }
    if (this.props.onClose) {
      this.props.onClose();
    }
  };

  updateFocus() {
    if (this.selectedRef && this.selectedRef.innerRef) {
      this.selectedRef.innerRef.focus();
    }
  }

  render() {
    const { children, open, focusedIndex, ...props } = this.props;
    let classes = MDC_SIMPLE_MENU;
    this.selectedRef = null;
    if (open) {
      classes += ` ${MDC_SIMPLE_MENU}--open`;
    }
    const element = (
      <div
        className={classes}
        tabIndex="-1"
        ref={(c) => {
          this.innerRef = c;
        }}
      >
        <ul
          className={`${MDC_SIMPLE_MENU}__items mdc-list`}
          role="menu"
          aria-hidden="true"
        >
          {Children.map(children, (child, index) => {
            if (child.props.mdcElement === "mdc-list-item") {
              const { tabIndex } = child.props;
              const p = {};
              p.onSelected = (c, i) => {
                this.handleOnSelected(c, index);
                if (child.props.onSelected) {
                  child.props.onSelected(c, i);
                }
              };

              if (focusedIndex === index) {
                p.selected = true;
                p.ref = (c) => {
                  this.selectedRef = c;
                };
              }
              if (tabIndex > -1) {
                return React.cloneElement(child, p);
              }
            }
            return child;
          })}
        </ul>
      </div>
    );
    return Zrmc.render(element, props);
  }
}

Menu.defaultProps = {
  mdcElement: MDC_SIMPLE_MENU,
  open: false,
  onSelected: null,
  onClose: null,
  anchorCorner: null,
  anchorMargin: null,
  focusedIndex: -1,
};

Menu.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node.isRequired,
  open: PropTypes.bool,
  onSelected: PropTypes.func,
  onClose: PropTypes.func,
  anchorCorner: PropTypes.oneOf([
    "top_start",
    "top_end",
    "bottom_start",
    "bottom_end",
  ]),
  anchorMargin: PropTypes.shape({
    top: PropTypes.string,
    left: PropTypes.string,
    bottom: PropTypes.string,
    right: PropTypes.string,
  }),
  focusedIndex: PropTypes.number,
};
