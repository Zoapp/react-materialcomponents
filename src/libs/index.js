/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Rmdc from "./rmdc";
import Button from "./components/button";
import Icon from "./components/icon";
import Content from "./components/content";
import Checkbox from "./components/checkbox";
import Fab from "./components/fab";
import TextField from "./components/textfield";
import Radio from "./components/radio";
import FormField from "./components/formField";
import Select from "./components/select";
import { Menu, MenuAnchor, MenuItem, MenuDivider } from "./menu";
import Snackbar from "./components/snackbar";
import LinearProgress from "./components/linearProgress";
import Switch from "./components/switch";
import Slider from "./components/slider";
import IconToggle from "./components/iconToggle";
import { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon } from "./toolbar";
import { Drawer, DrawerContent, DrawerHeader } from "./drawer";
import { List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader } from "./list";
import { Grid, Inner, Cell } from "./layoutGrid";
import { Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle } from "./cards";
import { Tabbar, Tab } from "./tab";
import { GridList, Tile } from "./gridList";
import { Dialog, DialogHeader, DialogBody, DialogFooter } from "./dialog";

export {
  Button, Content, Fab, TextField, Checkbox, Radio, FormField, Select, Snackbar,
  LinearProgress, Switch, Slider, IconToggle, Icon,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerContent, DrawerHeader,
  List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader,
  Grid, Inner, Cell,
  Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle,
  Tabbar, Tab,
  GridList, Tile,
  Dialog, DialogHeader, DialogBody, DialogFooter,
  Menu, MenuAnchor, MenuItem, MenuDivider,
};

export default Rmdc;
