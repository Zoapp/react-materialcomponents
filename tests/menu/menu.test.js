import React from "react";
import renderer from "react-test-renderer";
import { Menu } from "libs/menu";

describe("menu/Menu", () => {
  it("can render it self correctly", () => {
    const tree = renderer
      .create(
        <Menu>
          <div>Children are here!</div>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be render open", () => {
    const tree = renderer
      .create(
        <Menu open>
          <div>Children are here!</div>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be render sticked to a specified corner", () => {
    const tree = renderer
      .create(
        <Menu anchorCorner="top_start">
          <div>Children are here!</div>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have an specified margins", () => {
    const tree = renderer
      .create(
        <Menu anchorMargin={{ top: "10" }}>
          <div>Children are here!</div>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have an index render focused", () => {
    const tree = renderer
      .create(
        <Menu focusedIndex={2}>
          <div>Children are here!</div>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
