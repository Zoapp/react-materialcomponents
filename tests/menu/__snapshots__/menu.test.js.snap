// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu/Menu can be render open 1`] = `
<div
  className="mdc-menu mdc-menu--open"
  tabIndex="-1"
>
  <ul
    aria-hidden="true"
    className="mdc-menu__items mdc-list"
    role="menu"
  >
    <div>
      Children are here!
    </div>
  </ul>
</div>
`;

exports[`menu/Menu can be render sticked to a specified corner 1`] = `
<div
  className="mdc-menu"
  tabIndex="-1"
>
  <ul
    aria-hidden="true"
    className="mdc-menu__items mdc-list"
    role="menu"
  >
    <div>
      Children are here!
    </div>
  </ul>
</div>
`;

exports[`menu/Menu can have an index render focused 1`] = `
<div
  className="mdc-menu"
  tabIndex="-1"
>
  <ul
    aria-hidden="true"
    className="mdc-menu__items mdc-list"
    role="menu"
  >
    <div>
      Children are here!
    </div>
  </ul>
</div>
`;

exports[`menu/Menu can have an specified margins 1`] = `
<div
  className="mdc-menu"
  tabIndex="-1"
>
  <ul
    aria-hidden="true"
    className="mdc-menu__items mdc-list"
    role="menu"
  >
    <div>
      Children are here!
    </div>
  </ul>
</div>
`;

exports[`menu/Menu can render it self correctly 1`] = `
<div
  className="mdc-menu"
  tabIndex="-1"
>
  <ul
    aria-hidden="true"
    className="mdc-menu__items mdc-list"
    role="menu"
  >
    <div>
      Children are here!
    </div>
  </ul>
</div>
`;
